pkg/kubelet/dockertools/manager.go

```diff
@@ -1212,32 +1215,43 @@ func (dm *DockerManager) runContainerInPod(pod *api.Pod, container *api.Containe
 		glog.Errorf("Failed to create symbolic link to the log file of pod %q container %q: %v", podFullName, container.Name, err)
 	}
 
-	// Set OOM score of POD container to lower than those of the other containers
-	// which have OOM score 0 by default in the pod. This ensures that it is
-	// killed only as a last resort.
-	containerInfo, err := dm.client.InspectContainer(string(id))
+	// Set OOM score of the container based on the priority of the container.
+	// Some containers are part of top-tier pods, and some containers are part of best-effort pods.
+	// Processes in best-effort pods should be killed first if the system runs out of memory.
+	// The main pod infrastructure container should be the last to get killed.
+	var oomScoreAdj int
+	if container.Name == PodInfraContainerName {
+		oomScoreAdj = qos.PodInfraOomAdj
+	} else {
+		oomScoreAdj = qos.GetPodOomAdjust(&pod.Spec)
+	}
+	fsManager := fs.Manager{
+		Cgroups: &configs.Cgroup{
```

Comment:
I'm not sure this will work for all cases. We should get the cgroup from Docker somehow (cAdvisor does this today, maybe we can piggyback off of it)